plan/dag_plan_test.go

```diff
@@ -136,3 +136,136 @@ func (s *testPlanSuite) TestDAGPlanBuilderSimpleCase(c *C) {
 		c.Assert(ToString(p), Equals, tt.best, Commentf("for %s", tt.sql))
 	}
 }
+
+func (s *testPlanSuite) TestDAGPlanBuilderBasePhysicalPlan(c *C) {
+	UseDAGPlanBuilder = true
+	defer func() {
+		UseDAGPlanBuilder = false
+		testleak.AfterTest(c)()
+	}()
+	tests := []struct {
+		sql  string
+		best string
+	}{
+		// Test for update.
+		{
+			sql: "select * from t order by b limit 1 for update",
+			// TODO: This is not reasonable. Mysql do like this because the limit of InnoDB, should TiDB keep consistency with MySQL?
+			best: "TableReader(Table(t))->Lock->Sort + Limit(1) + Offset(0)",
+		},
+		// Test complex update.
+		{
+			sql:  "update t set a = 5 where b < 1 order by d limit 1",
+			best: "TableReader(Table(t)->Sel([lt(test.t.b, 1)])->Limit)->Limit->*plan.Update",
```

Comment:
There is a bug in old planner ...

mysql> create table t1(a int primary key,b int);
Query OK, 0 rows affected (0.01 sec)

mysql> explain update t1 set a = 1 where a < 1 order by b limit 1;
+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| ID          | Json                                                                                                                                                                                                                                                                                            | ParentID |
+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
| TableScan_6 | {
    "db": "test",
    "table": "t1",
    "desc": false,
    "keep order": true,
    "push down info": {
        "limit": 1,
        "access conditions": [
            "lt(test.t1.a, 1)"
        ],
        "index filter conditions": null,
        "table filter conditions": null
    }
} | Update_5 |
| Update_5    | {
    "children": [
        "TableScan_6"
    ]
}                                                                                                                                                                                                                                               |          |
+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+
2 rows in set (0.00 sec)